packages/algob/src/internal/deployer.ts

```diff
@@ -228,13 +241,31 @@ class DeployerBasicMode {
   }
 
   /**
-   * Returns cached program (from artifacts/cache) `ASCCache` object by filename.
-   * TODO: beta support - this will change
-   * @param name ASC name used during deployment
+   * Returns cached program (from artifacts/cache) `ASCCache` object by app/lsig name.
+   * @param name App/Lsig name used during deployment
    */
-  getDeployedASC (name: string): Promise<ASCCache | undefined> {
+  async getDeployedASC (name: string): Promise<ASCCache | AppCache | undefined> {
     const op = new CompileOp(this.algoOp.algodClient);
-    return op.readArtifact(name);
+
+    // app
+    const app = this.getAppByName(name);
+    if (app !== undefined) {
+      const approvalCache = await op.readArtifact(app.approvalFile);
+      const clearCache = await op.readArtifact(app.clearFile);
+      return {
+        approval: approvalCache,
+        clear: clearCache
+      };
+    }
+
+    // lsig
+    const resultMap = this.cpData.precedingCP[this.networkName]?.dLsig ?? new Map();
+    const lsigInfo = resultMap.get(name);
+    if (lsigInfo?.file) {
+      return await op.readArtifact(lsigInfo.file);
+    }
+
+    return undefined;
```

Comment:
maybe we should throw exception instead of undefined?

Reply:
it's just that whenever we get data from checkpoint we return `undefined` if not found. Similar to `getDelegatedLsig`, `getContractLsig`, `getLsigByName`, `getAppByName`, `getApp`. 

Should we update all of them?